### PR TITLE
separate transfer queue to active and standby processor

### PR DESCRIPTION
**Pull request by wxing1292**
separate queue ack mgr into separate file

Review #648 first, since this PR contains some code from #648

UT in next PR

### CODE REVIEW

File: service/history/replicatorQueueProcessor.go [hunk 1]
```diff
@@ -99,6 +109,8 @@ func (p *replicatorQueueProcessorImpl) Process(qTask queueTaskInfo) error {
 
 	if err != nil {
 		p.metricsClient.IncCounter(scope, metrics.TaskRequests)
```
Comment (samarabbas) [inline service/history/replicatorQueueProcessor.go]: Can you change this to TaskFailure?

File: service/history/replicatorQueueProcessor.go [hunk 2]
```diff
@@ -99,6 +109,8 @@ func (p *replicatorQueueProcessorImpl) Process(qTask queueTaskInfo) error {
...
+	} else {
+		p.queueAckMgr.completeTask(task.TaskID)
```
Comment (samarabbas) [inline service/history/replicatorQueueProcessor.go]: Why move this out of base processor?

File: service/history/replicatorQueueProcessor.go (hunk 1 above)
Comment (wxing1292) [reply service/history/replicatorQueueProcessor.go]: sure

File: service/history/historyEngineInterfaces.go [hunk 3]
```diff
@@ -118,12 +132,11 @@ type (
 	}
 
 	timerQueueAckMgr interface {
-		readRetryTimerTasks() []*persistence.TimerTaskInfo
+		getFinishedChan() <-chan struct{}
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply service/history/historyEngineInterfaces.go]: plz ignore the timer* files

File: service/history/replicatorQueueProcessor.go (hunk 2 above)
Comment (wxing1292) [reply service/history/replicatorQueueProcessor.go]: i believe there should be some logical separation of base processor and actual processor.
the base processor should only deal with retry, when seeing a error, and the actual task processor (the actual processing logic) should deal with errors such as entity not exists.

File: service/history/historyEngineInterfaces.go [hunk 4]
```diff
@@ -88,21 +93,30 @@ type (
...
+		readTasks(readLevel int64) ([]queueTaskInfo, bool, error)
+		completeTask(taskID int64) error
+		updateAckLevel(taskID int64) error
 	}
 
 	transferQueueProcessor interface {
 		common.Daemon
-		NotifyNewTask()
+		NotifyNewTask(clusterName string)
+		SetCurrentTime(clusterName string, currentTime time.Time)
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply service/history/historyEngineInterfaces.go]: this is delay the processing of the standby transfer tasks.

**Pull request merged**